linera-service/src/cli_wrappers/local_kubernetes_net.rs

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    docker::Docker, helm::Helm, kind::KindCluster, kubectl::Kubectl, util::get_github_root,
+};
+use crate::{
+    cli_wrappers::{ClientWrapper, LineraNet, LineraNetConfig, Network},
+    util::{self, current_binary_parent, CommandExt},
+};
+use anyhow::{anyhow, bail, Result};
+use async_trait::async_trait;
+use k8s_openapi::api::core::v1::Pod;
+use kube::{
+    api::{Api, ListParams},
+    Client,
+};
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+use tempfile::{tempdir, TempDir};
+use tokio::process::Command;
+
+/// The information needed to start a [`LocalKubernetesNet`].
+pub struct LocalKubernetesNetConfig {
+    pub network: Network,
+    pub testing_prng_seed: Option<u64>,
+    pub binaries_dir: Option<PathBuf>,
+    pub cluster_id: Option<u32>,
+}
+
+/// A set of Linera validators running locally as native processes.
+pub struct LocalKubernetesNet {
+    network: Network,
+    testing_prng_seed: Option<u64>,
+    next_client_id: usize,
+    tmp_dir: Arc<TempDir>,
+    binaries_dir: Option<PathBuf>,
+    kubectl: Option<Kubectl>,
+    kind_cluster: Option<KindCluster>,
+    cluster_id: Option<u32>,
+}
+
+#[async_trait]
+impl LineraNetConfig for LocalKubernetesNetConfig {
+    type Net = LocalKubernetesNet;
+
+    async fn instantiate(self) -> Result<(Self::Net, ClientWrapper)> {
+        let mut net = LocalKubernetesNet::new(
+            self.network,
+            self.testing_prng_seed,
+            self.binaries_dir,
+            self.cluster_id,
+        )?;
+        let client = net.make_client();
+        net.generate_initial_validator_config().await.unwrap();
+        client.create_genesis_config().await.unwrap();
+        net.run().await.unwrap();
+        Ok((net, client))
+    }
+}
+
+#[async_trait]
+impl LineraNet for LocalKubernetesNet {
+    async fn ensure_is_running(&mut self) -> Result<()> {
+        let client = Client::try_default().await?;
+        let pods: Api<Pod> = Api::namespaced(client, "default");
+
+        let list_params = ListParams::default().labels("app=proxy");
+        for pod in pods.list(&list_params).await? {
+            if let Some(status) = pod.status {
+                if let Some(phase) = status.phase {
+                    if phase != "Running" {
+                        bail!(
+                            "Validator {} is not Running",
+                            pod.metadata
+                                .name
+                                .expect("Fetching pod name should not fail")
+                        );
+                    }
+                }
+            }
+        }
+
+        let lp = ListParams::default().labels("app=shards");
+        for p in pods.list(&lp).await? {
```

Comment:
Please fix the variable names

Reply:
Wtf this is supposed to be fixed, I fixed it on `v4`:

![Screenshot 2023-11-16 at 17.13.14.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/HlciHFAoHZW62zn13apJ/6900e091-22b7-4a45-9d33-cec9e612fdaf.png)

Anyways, it's been fixed locally, so hopefully once I update this it'll show the right code